synapse/handlers/room.py

```diff
@@ -427,6 +588,30 @@ def send(etype, content, **kwargs):
                 content=content,
             )
 
+    @defer.inlineCallbacks
+    def _generate_room_id(self, creator_id, is_public):
+        # autogen room IDs and try to create it. We may clash, so just
+        # try a few times till one goes through, giving up eventually.
+        attempts = 0
+        while attempts < 5:
+            try:
+                random_string = stringutils.random_string(18)
+                gen_room_id = RoomID(
+                    random_string,
+                    self.hs.hostname,
+                ).to_string()
+                if isinstance(gen_room_id, bytes):
+                    gen_room_id = gen_room_id.decode('utf-8')
```

Comment:
How on earth can we end up with `bytes` here?

Reply:
`DomainSpecificString.to_string` returns a `str`, which is a `bytes` under python 2.